### PR TITLE
chore(Example): Documented @ngrx/entity and added to example app

### PR DESCRIPTION
**Pull request by brandonroberts**

### CODE REVIEW

File: docs/entity/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+Entity State adapter for managing record collections.
+
+@ngrx/entity provides an API to manipulate and query entity collections.
+
+- Reduces boilerplate for managing common datasets
+- Provides performant operations for managing entity collections
+- Extensible type-safe adapters for selecting entity information
+
+### Installation
+Install @ngrx/effects from npm:
```
Comment (mhheise) [inline docs/entity/README.md]: Install `@ngrx/effects` -> Install `@ngrx/entity`?

Comment (brandonroberts) [reply docs/entity/README.md]: Yep, thanks

File: docs/entity/adapter.md [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+`user.reducer.ts`
+```ts
+import * as user from './user.actions';
+
+export interface State extends EntityState<User> {
+  // additional entities state properties
+  selectedUserId: number | null;
+}
+
+export const adapter: EntityAdapter<User> = createEntityAdapter<User>({
```
Comment (peterbsmith2) [inline docs/entity/adapter.md]: ought to be userAdaptor?

Comment (peterbsmith2) [reply docs/entity/adapter.md]: or line 191 ought to be adapter?

File: docs/entity/adapter.md [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+export const UPDATE_USER = '[User] Update User';
+export const UPDATE_USERS = '[User] Update Users';
+export const DELETE_USER = '[User] Delete User';
+export const DELETE_USERS = '[User] Delete Users';
+export const CLEAR_USERS = '[User] Clear Users';
+
+export class LoadUsers implements Action {
+  readonly type = LOAD_USERS;
+
+  constructor(public payload: { user: User[] }) {}
```
Comment (diginikkari) [inline docs/entity/adapter.md]: Should this payload be users?

File: docs/entity/adapter.md [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+export class UpdateUser implements Action {
+  readonly type = UPDATE_USER;
+
+  constructor(public payload: { user: User }) {}
+}
+
+export class UpdateUsers implements Action {
+  readonly type = UPDATE_USERS;
+
+  constructor(public payload: { user: User[] }) {}
```
Comment (diginikkari) [inline docs/entity/adapter.md]: Should this payload also be users?

File: docs/entity/adapter.md [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+export class DeleteUser implements Action {
+  readonly type = DELETE_USER;
+
+  constructor(public payload: { user: User }) {}
+}
+
+export class DeleteUsers implements Action {
+  readonly type = DELETE_USERS;
+
+  constructor(public payload: { user: User[] }) {}
```
Comment (diginikkari) [inline docs/entity/adapter.md]: Should this also be users

**Pull request merged**